### ISSUE TITLE
Add error.expected and error.actual to spec reporter failure output

### ISSUE THREAD

**Issue by vgrigoruk** [webdriverio/webdriverio#5035]
**Is your feature request related to a problem? Please describe.**
At the moment, `@wdio/spec-reporter` ignores `error.actual` and `error.expected` fields (which are usually set on `assert.AssertionError` by packages like `chai`). These properties are not included in reporter output, which makes test failure analysis time-consuming.
This could also be useful for other reporters (in particular, `@wdio/junit-reporter`).

**Describe the solution you'd like**
Add something like this before this line:  https://github.com/webdriverio/webdriverio/blob/master/packages/wdio-spec-reporter/src/index.js#L283

```js
if (error.actual || error.expected) {
  output.push(this.chalk.green(error.expected))
  output.push(this.chalk.red(error.actual))
}

```

**Describe alternatives you've considered**

**Additional context**

How this is done in MochaJS: https://mochajs.org/#diffs

**Comment by christian-bromann** [comment]
@vgrigoruk I would say formatting the error message is task of the assertion library. I know that `expect-webdriverio` (which will be build into the test runner) is already providing that:

![toHaveText](https://github.com/webdriverio/expect-webdriverio/blob/master/docs/img/errors/text.png?raw=true "toHaveText")

It seems that building this into WebdriverIO natively is a lot of work given all the reporters that exist.

**Comment by vgrigoruk** [comment]
It is very nice to finally have a built-in expectations in webdriverio, however I disagree this is a task of assertion library. Here is how both `mocha` and `jasmine` work in default configuration with node built-in `AssertionError` with the following test:

```js
const { AssertionError } = require('assert');

describe('test', () => {
  it('should fail', () => {
    throw new AssertionError({ message: 'boom', expected: 'expected', actual: 'actual'});
  })
})
```

Mocha:
```
> npx mocha



  test
    1) should fail


  0 passing (5ms)
  1 failing

  1) test
       should fail:

      AssertionError [ERR_ASSERTION]: boom
      + expected - actual

      -actual
      +expected
      
      at new AssertionError (internal/assert/assertion_error.js:418:11)
      at Context.<anonymous> (test.js:5:11)
      at processImmediate (internal/timers.js:439:21)
```

Jasmine:

```
> npx jasmine test.js

Randomized with seed 55544
Started
F

Failures:
1) test should fail
  Message:
    AssertionError: boom
  Stack:
    error properties: Object({ generatedMessage: false, code: 'ERR_ASSERTION', actual: 'actual', expected: 'expected', operator: undefined })
        at <Jasmine>
        at new AssertionError (internal/assert/assertion_error.js:418:11)
        at UserContext.<anonymous> (/Users/vgrygoruk/code/test-assert-jasmine-mocha/test.js:5:11)
        at <Jasmine>
        at processImmediate (internal/timers.js:439:21)

1 spec, 1 failure
Finished in 0.007 seconds
```

**Comment by dpoetzsch** [comment]
This would be really helpful. Is there a way I can help to get this done?

**Comment by christian-bromann** [comment]
> Is there a way I can help to get this done?

@dpoetzsch it would be great if we could support this for our spec reporter for the case the user is not using the built in assertion library. Would you like to pick this up?

**Comment by apkatsikas** [comment]
@christian-bromann it also looks like the report itself does not show a simple assertion failure when using the new expect lib - is this a separate issue or related? instead it throws "CompoundError"

![image](https://user-images.githubusercontent.com/24813300/80238253-165b8100-862c-11ea-8f45-e10c36e6f2bd.png)

**Comment by christian-bromann** [comment]
@apkatsikas yes, I am afraid this is specific to Allure

**Comment by apkatsikas** [comment]
> @apkatsikas yes, I am afraid this is specific to Allure

ok @christian-bromann i will watch this space. let me know if you require more info

**Comment by christian-bromann** [comment]
@apkatsikas feel free to help contribute a bug fix if you are interested in this

**Comment by christian-bromann** [comment]
implemented in #6943

**Comment by torbenkohlmeier** [comment]
@apkatsikas Since the latest release, the CompoundError should include the error message which contains details about the assertion error (#12378).
